plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -23,14 +23,16 @@ function perflab_query_plugin_info( string $plugin_slug ) {
 	$plugins       = get_transient( $transient_key );
 
 	if ( is_array( $plugins ) ) {
-		// If the specific plugin_slug is not in the cache, return an error.
-		if ( ! isset( $plugins[ $plugin_slug ] ) ) {
-			return new WP_Error(
-				'plugin_not_found',
-				__( 'Plugin not found in cached API response.', 'performance-lab' )
-			);
+		if ( isset( $plugins[ $plugin_slug ] ) ) {
+			if ( false === $plugins[ $plugin_slug ] ) {
+				// Plugin was requested before and not found.
+				return new WP_Error(
+					'plugin_not_found',
+					__( 'Plugin not found in cached API response.', 'performance-lab' )
+				);
+			}
```

Comment:
If `$plugins[ $plugin_slug ]` is either an `array` or a `WP_Error`, then this can just return that.

```suggestion
```